sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -1353,62 +1393,64 @@ pi_result _pi_queue::executeCommandList(pi_command_list_ptr_t CommandList,
     KernelsToBeSubmitted.clear();
   }
 
-  // In this mode all inner-batch events have device visibility only,
-  // and we want the last command in the batch to signal a host-visible
-  // event that anybody waiting for any event in the batch will
-  // really be using.
-  //
-  if (EventsScope == LastCommandInBatchHostVisible) {
-    // Create a "proxy" host-visible event.
+  if (!UseImmediateCommandLists) {
```

Comment:
This "if" shouldn't be needed. It is invalid setting to have EventsScope == LastCommandInBatchHostVisible with immediate submission. Please just add an assert instead.

Reply:
OK

Reply:
Actually, that if guards more than just the (EventsScope == LastCommandInBatchHostVisible). Some other actions below that are also to be done only in bacthed mode.